src/lib/pmd/PmdCatalogWrapper.ts

```diff
@@ -84,17 +84,64 @@ export class PmdCatalogWrapper extends PmdSupport {
     return JSON.parse(rawCatalog.toString());
   }
 
-  protected buildCommandArray(): [string, string[]] {
+  protected async buildCommandArray(): Promise<[string, string[]]> {
     const command = 'java';
     // NOTE: If we were going to run this command from the CLI directly, then we'd wrap the classpath in quotes, but this
     // is intended for child_process.spawn(), which freaks out if you do that.
-    const args = ['-cp', this.buildClasspath().join(':'), MAIN_CLASS, PMD_LIB, PMD_VERSION, SUPPORTED_LANGUAGES.join(',')];
+    const classpathEntries = await this.buildClasspath();
+    const parameters = await this.buildCatalogerParameters();
+    const args = ['-cp', classpathEntries.join(':'), MAIN_CLASS, ...parameters];
     return [command, args];
   }
 
-  protected buildClasspath(): string[] {
+  protected async buildClasspath(): Promise<string[]> {
     // TODO: This probably isn't where the JAR ought to live. Once the JAR's home is finalized, come back to this.
     const catalogerLibs = `${PMD_CATALOGER_LIB}/*`;
-    return super.buildClasspath().concat([catalogerLibs]);
+    const classpathEntries = await super.buildClasspath();
+    classpathEntries.push(catalogerLibs);
+    return classpathEntries;
+  }
+
+  private async buildCatalogerParameters(): Promise<string[]> {
```

Comment:
Crux of the change is here. This portion of the code fetches the custom rule path entries and merges PMD's inbuilt rules into the same Map. It then builds each argument as language=/rule/path,/rule/path2